src/components/data-pane/type-changer.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import * as CSSModules from 'react-css-modules';
+
+import { ExpandedType } from 'compassql/build/src/query/expandedtype';
+import {DATASET_SCHEMA_CHANGE_FIELD_TYPE, DatasetSchemaChangeFieldType} from '../../actions/dataset';
+import {ActionHandler} from '../../actions/redux-action';
+import {ShelfFieldDef} from '../../models/shelf/encoding';
+import * as styles from './type-changer.scss';
+
+export interface TypeChangerProps extends ActionHandler<DatasetSchemaChangeFieldType> {
+  fieldDef: ShelfFieldDef;
+  types: ExpandedType[];
+}
+
+export interface TypeChangerState {
```

Comment:
I wonder if we really need internal state for selectedType here.  
If I understand correctly, I think we should not have it as we can directly read the input `type` props.